### PR TITLE
move addDefaultVersionIdentifiers to target

### PR DESCRIPTION
**Pull request by lucica28**
The purpose of this pr is to improve dmd lib interface. For my particular use case, I am using methods inside `frontend.d`, but `frontend.d` imports `mars.d`, which should be simply the entry point, so it can be omitted when building other projects that use `dmd-as-a-library`

### CODE REVIEW

File: compiler/src/dmd/target.d [hunk 1]
```diff
@@ -1141,7 +1325,7 @@ extern (C++) struct Target
      * Returns:
      *  alignment of the stack
      */
-    extern (D) uint stackAlign()
+    extern (C++) uint stackAlign()
```
Comment (dkorpel) [inline compiler/src/dmd/target.d:1328]: Why this change?

**Review by dkorpel — COMMENTED** [review]
Is the move a copy paste or are there additional changes?

File: compiler/src/dmd/target.d (hunk 1 above)
Comment (lucica28) [reply compiler/src/dmd/target.d]: It is indeed a copy paste + changing some imports, this might have slipped as this is not an intended change :)

**Review by dkorpel — APPROVED** [review]

**Review by RazvanN7 — APPROVED** [review]
